pkg/provider/akeyless/akeyless_api.go

```diff
@@ -82,6 +84,7 @@ func (a *akeylessBase) GetToken(accessID, accType, accTypeParam string, k8sAuth
 
 func (a *akeylessBase) GetSecretByType(ctx context.Context, secretName, token string, version int32) (string, error) {
 	item, err := a.DescribeItem(ctx, secretName, token)
+	metrics.ObserveAPICall(constants.ProviderAKEYLESSSM, constants.CallAKEYLESSSMDescribeSecret, err)
```

Comment:
This `.ObserveAPICall()` is being called twice: once inside `a.DescribeItem()` and right here. I'd suggest to remove it here and just call it in `DescribeItem()`.